throw-money-app/src/pages/Dashboard.jsx

```diff
@@ -1,21 +1,105 @@
+
+Modal.setAppElement('#root')
 
 const Dashboard = () => {
   const dispatch = useDispatch()
   const selector = useSelector((state) => state)
   // state保持データ確認用
+  const uid = selector.users.uid
   const userName = selector.users.userName
   const remainMoney = selector.users.remainMoney
 
+  // モーダル制御
+  const [modalIsOpen, setMpdalIsOpen] = useState(false)
+
+  // walletクリックイベント
+  // モーダルコンポーネントを返却
+  const handleClickWallet = (otherUserName, otherUserWallet) => {
+    setMpdalIsOpen(true)
+    return (
+      <Modal isOpen={modalIsOpen} style={modalStyle} onRequestClose={() => setMpdalIsOpen(false)}>
+        <p>ユーザ名：{otherUserName}</p>
+        <p>残高：{otherUserWallet}</p>
+        <p>
+          <button onClick={() => setMpdalIsOpen(false)}>閉じる</button>
+        </p>
+      </Modal>
+    )
+  }
+
+  // 他ユーザ情報
+  const [otherUsersInfo, setOtherUsersInfo] = useState([])
+  console.log('ダッシュボードコンポーネント otherUsersInfo:' + otherUsersInfo)
+
+  // 他ユーザ情報を取得する
+  // 第２引数にotherUsersInfoをセット (otherUsersInfoが変更された場合に実行)
+  useEffect(async () => {
+    // 他のuser情報の取得
+    console.log('ダッシュボード useEffect:')
+    const otherUsersRef = await db.collection('users').where('uid', '!=', uid).get()
+    // ダッシュボード内で表示用に成型(firebaseからの返却値ではmap関数が使えないため)
+    const otherUsersInfoTmp = []
+    otherUsersRef.forEach((doc) => {
+      const otherUserData = doc.data()
+      otherUsersInfoTmp.push([
+        {
+          id: otherUserData.uid,
+          otherUserName: otherUserData.username,
+          otherUserWallet: otherUserData.remainMoney,
+        },
+      ])
+    })
+
+    //stateにセット
+    setOtherUsersInfo(otherUsersInfoTmp)
+  }, [])
+
   return (
     <div>
       <h2>ダッシュボード</h2>
       <p>{userName}さん、ようこそ</p>
       <p>残高：{remainMoney}</p>
 
+      <h3>ユーザ一覧</h3>
+      <table>
+        <tbody>
+          <tr>
+            <th>ユーザ名</th>
+            <th>wallet</th>
+            <th></th>
+          </tr>
+          {/**他ユーザ情報の出力*/}
+          {otherUsersInfo.map((userData, index) => {
+            //console.log('userData:' + JSON.stringify(userData))
+            return (
+              <tr key={index}>
+                <td>{userData[0].otherUserName}</td>
+                <td>
+                  <button onClick={() => handleClickWallet(userData[0].otherUserName, userData[0].otherUserWallet)}>
+                    walletを見る: {userData[0].otherUserWallet}
+                  </button>
+                  {/* <Modal isOpen={modalIsOpen} style={modalStyle} onRequestClose={() => setMpdalIsOpen(false)}>
+                    <p>ユーザ名：{userData[0].otherUserName}</p>
+                    <p>残高：{userData[0].otherUserWallet}</p>
+                    <p>
+                      <button onClick={() => setMpdalIsOpen(false)}>閉じる</button>
+                    </p>
+                  </Modal> */}
+                </td>
+                <td>送る</td>
+              </tr>
+            )
+          })}
+        </tbody>
+      </table>
+
```

Comment:
以下のようにレンダリングさせておいて、isOpenの真偽値を変更すると表示非表示ができるような設計のイメージです。
値が渡っているのでしたらあとは表示の切り替えができれば意図した挙動になると思うので、試してみていただきたいです。
```suggestion
<Modal isOpen={modalIsOpen} style={modalStyle} onRequestClose={() => setMpdalIsOpen(false)} />
```
まずはisOpenの真偽値を直接（ハードコーディングして）切り替えてみて、表示の切り替えができるかどうかかから試してみると良いと思います。

Reply:
提示頂いた内容を理解しきれていないかもしれませんが、１０２行に上記コードを追加しただけでは期待通りの動きにはならずでして。（isOpenをtrueにするだけでは、モーダルの中身は表示されない。実装しながら、そうなるかなとは思いましたが）
ちなみに、`handleClickWallet `関数には特に手を加えない認識ですが相違なかったでしょうか。

（再送：前回の質問内容）
まずreact-modalは、モーダルで表示させたい情報を<Modal></Modal>で囲んで記載するものだと理解しています。
それを`handleClickWallet `関数でやろうとしていますが、値は渡っているがモーダルは表示されない結果となりました。
現状その原因はreact-modalを使っているからと考えていて、その認識を前回確認させて頂きました。
（`handleClickWallet `に改修の余地があれば指摘頂きたいです）

Reply:
>提示頂いた内容を理解しきれていないかもしれませんが、１０２行に上記コードを追加しただけでは期待通りの動きにはならずでして。（isOpenをtrueにするだけでは、モーダルの中身は表示されずです。そうなるかなとは想定していましたが）

isOpenをtrueにしたら、空のモーダルが表示されるということであっておりますか？
前回、値は渡せているとのことでしたので、空のモーダルが表示できたらあとは中身になる値を渡せば良い認識でしたがいかがでしょうか。

>まずreact-modalは、モーダルで表示させたい情報を<Modal></Modal>で囲んで記載するものだと理解しています。

はい、それでもできますが、propsでモーダルに渡したい値を記載すれば良いと思い省略系の記述をさせていただきました。
ドキュメントを見ると、dataという名前で渡せそうなのですがどうでしょうか？
http://reactcommunity.org/react-modal/

>それをhandleClickWallet 関数でやろうとしていますが、値は渡っているがモーダルは表示されない結果となりました。
現状その原因はreact-modalを使っているからと考えていて、

関数は確かにモーダルコンポーネントを返却するものですが、例えばですがモーダルではなくdivタグで囲った要素でしたら表示されるのでしょうか？
原因がreact-modalにあると思っている根拠があれば教えていただきたいです。

Reply:
最新コミットにて課題機能を実現できましたので、レビューをお願い致します。

>原因がreact-modalにあると思っている根拠があれば教えていただきたいです。

失礼しました。コンポーネントを返却するとレンダリングされるという、謎の思い込みをしておりました。
Reactのレンダリングはstateが更新された時、という大前提が抜け落ちていました。
ここから、再度`handleClickWallet `でモーダルで表示したい内容をstateにセット、<Modal>内でstateから値を取得するというイメージの繋がりまして、解決できたかなと思います。
基本大事だと身に染みた感がすごいです..お手数おかけしました(..)

>ドキュメントを見ると、dataという名前で渡せそうなのですがどうでしょうか？
http://reactcommunity.org/react-modal/

`<Modal data={otherUsersName} />`という感じでpropsを渡すイメージはついているのですが、
ユーザ情報とwalletをモーダルの中に表示させるためには、今回コミットしたように<Modal>で囲って直書きする方法しか思いつきませんでした。dataの使われ方がいまいち掴めない感じではありますが、今のところdataを使っての実装に拘るつもりはありません。
（<Modal>内が肥大化した場合は、react-modalが入っている自作コンポーネントを作り、そのコンポーネントにpropsを渡すような方法はイメージついていますが、これは単なるpropsのバケツリレーですよね）

１点質問なのですが、提示頂いた`http://reactcommunity.org/react-modal/`はどのように辿りつかれたのでしょうか。
トップページから、react-modalへのリンク等もなく`https://ja.reactjs.org/`などからも辿りつけずでして。